crates/services/txpool/src/service.rs

```diff
@@ -205,29 +215,38 @@ where
             new_transaction = self.gossiped_tx_stream.next() => {
                 if let Some(GossipData { data: Some(tx), message_id, peer_id }) = new_transaction {
                     let id = tx.id(&self.shared.consensus_params.chain_id);
-                    let txs = vec!(Arc::new(tx));
-                    let mut result = tracing::info_span!("Received tx via gossip", %id)
-                        .in_scope(|| {
-                            self.shared.txpool.lock().insert(
-                                &self.shared.tx_status_sender,
-                                &txs
-                            )
-                        });
-
-                    if let Some(acceptance) = match result.pop() {
-                        Some(Ok(_)) => {
-                            Some(GossipsubMessageAcceptance::Accept)
-                        },
-                        Some(Err(_)) => {
-                            Some(GossipsubMessageAcceptance::Reject)
+                    let current_height = self.shared.db.current_block_height()?;
+
+                    // verify tx
+                    let checked_tx = check_single_tx(tx, current_height, &self.shared.config).await;
+
+                    if let Ok(tx) = checked_tx {
```

Comment:
I guess in the case that it is not valid, we need to report `GossipsubMessageAcceptance::Reject`